packages/grid/x-data-grid-pro/src/hooks/features/lazyLoader/useGridLazyLoader.ts

```diff
@@ -0,0 +1,220 @@
+        props.rowsLoadingMode !== GridFeatureModeConstant.server ||
+        (renderedRowsIntervalCache.current.firstRowToRender === params.firstRowToRender &&
+          renderedRowsIntervalCache.current.lastRowToRender === params.lastRowToRender)
+      ) {
+        return;
+      }
+
+      const renderedRowsIds: Array<GridRowId | null> = [...rowIds].splice(
+        params.firstRowToRender,
+        params.lastRowToRender - params.firstRowToRender,
+      );
+      const hasSkeletonRowIds = renderedRowsIds.some((rowId) =>
+        `${rowId}`.includes(GRID_SKELETON_ROW_ROOT_ID),
+      );
+
+      if (!hasSkeletonRowIds) {
+        return;
+      }
+
+      const { firstRowIndex, lastRowIndex } = findSkeletonRowsSection(visibleRows.rows, {
+        firstRowIndex: params.firstRowToRender,
+        lastRowIndex: params.lastRowToRender,
+      });
+
+      renderedRowsIntervalCache.current = params;
+
+      const fetchRowsParams: GridFetchRowsParams = {
+        firstRowToRender: firstRowIndex,
+        lastRowToRender: lastRowIndex,
+        sortModel,
+        filterModel,
+      };
+
+      apiRef.current.publishEvent('fetchRows', fetchRowsParams);
+    },
+    [
+      apiRef,
+      props.rowsLoadingMode,
+      rowIds,
+      sortModel,
+      filterModel,
+      visibleRows,
+      props.experimentalFeatures,
+    ],
+  );
+
+  const handleGridSortModelChange = React.useCallback<GridEventListener<'sortModelChange'>>(
+    (newSortModel) => {
+      const dimensions = apiRef.current.getRootDimensions();
+
+      if (
+        !(props.experimentalFeatures as GridExperimentalProFeatures)?.lazyLoading ||
+        !dimensions ||
+        props.rowsLoadingMode !== GridFeatureModeConstant.server
+      ) {
+        return;
+      }
+
+      const currentRenderContext = apiRef.current.unstable_getRenderContext();
+      const [firstRowToRender, lastRowToRender] = getRenderableIndexes({
+        firstIndex: currentRenderContext.firstRowIndex,
+        lastIndex: currentRenderContext.lastRowIndex,
+        minFirstIndex: 0,
+        maxLastIndex: visibleRows.rows.length,
+        buffer: props.rowBuffer,
+      });
+      const fetchRowsParams: GridFetchRowsParams = {
+        firstRowToRender,
+        lastRowToRender,
+        sortModel: newSortModel,
+        filterModel,
+      };
+
+      apiRef.current.publishEvent('fetchRows', fetchRowsParams);
+    },
+    [
+      apiRef,
+      props.rowsLoadingMode,
+      props.rowBuffer,
+      visibleRows.rows,
+      filterModel,
+      props.experimentalFeatures,
+    ],
+  );
+
+  const handleGridFilterModelChange = React.useCallback<GridEventListener<'filterModelChange'>>(
+    (newFilterModel) => {
+      const dimensions = apiRef.current.getRootDimensions();
+
+      if (
+        !(props.experimentalFeatures as GridExperimentalProFeatures)?.lazyLoading ||
+        !dimensions ||
+        props.rowsLoadingMode !== GridFeatureModeConstant.server
+      ) {
+        return;
+      }
```

Comment:
It's a bit complex to read all these conditions. You could split them:

```suggestion
      const { lazyLoading } = (props.experimentalFeatures ?? {}) as GridExperimentalProFeatures;

      if (!lazyLoading || !dimensions) {
        return;
      }

      if (props.rowsLoadingMode !== GridFeatureModeConstant.server) {
        return;
      }

      if (
        renderedRowsIntervalCache.current.firstRowToRender === params.firstRowToRender &&
        renderedRowsIntervalCache.current.lastRowToRender === params.lastRowToRender
      ) {
        return;
      }
```

The two first `if` could even be moved to a function and reused in other places of this hook.